python/lsst/ts/mtdome/power_management/slip_ring.py

```diff
@@ -0,0 +1,246 @@
+            power_over_limit_fraction = 0.0
+        else:
+            power_over_limit_fraction = (
+                self.max_power_drawn - CONTINUOUS_SLIP_RING_POWER_CAPACITY
+            ) / OVER_LIMIT_POWER_AVAILABLE
+        time_over_limit_fraction = self.time_over_limit / MAXIMUM_OVER_LOW_LIMIT_TIME
+        return (
+            MAXIMUM_COOL_DOWN_TIME
+            * power_over_limit_fraction
+            * time_over_limit_fraction
+        )
+
+    async def get_available_power(self, current_power_draw: float) -> float:
+        """Get the available power.
+
+        Determine whether the maximum power is available or none at all based
+        on the current power draw and, if the power draw has been over the
+        continuous limit, for how long. If the slip ring is disabled, the
+        returned value will be zero.
+
+        Parameters
+        ----------
+        current_power_draw : `float`
+            The amount of power currently in use [W].
+
+        Returns
+        -------
+        float
+            The available power [W].
+        """
+        if self.disabled:
+            return await self._handle_disabled()
+        match self.state:
+            case SlipRingState.BELOW_LOW_LIMIT:
+                return await self._handle_below_low_mimit(current_power_draw)
+            case SlipRingState.OVER_LOW_LIMIT:
+                return await self._handle_over_low_mimit(current_power_draw)
+            case SlipRingState.COOLING_DOWN:
+                return await self._handle_cooling_down()
+            case _:
+                raise ValueError(
+                    f"encountered unknown SlipRingState {self.state.name}."
+                )
+
+    async def _handle_disabled(self) -> float:
+        """Handle the DISABLED case.
+
+        Returns
+        -------
+        float
+            The available power [W]. In this case, always 0.0 is returned.
+        """
+        return 0.0
+
+    async def _handle_below_low_mimit(self, current_power_draw: float) -> float:
+        """Handle the BELOW_LOW_LIMIT case.
+
+        Transit to OVER_LOW_LIMIT if the power drawn is higher than the low
+        limit.
+
+        Parameters
+        ----------
+        current_power_draw : `float`
+            The amount of power currently in use [W].
+
+        Returns
+        -------
+        float
+            The available power [W].
+        """
+        if current_power_draw > CONTINUOUS_SLIP_RING_POWER_CAPACITY:
+            # Keep track of the maximum drawn power for calculating the cool
+            # down time.
+            self.max_power_drawn = max(self.max_power_drawn, current_power_draw)
+            # When the power draw started exceeding the continuous limit?
+            self.over_limit_start_time = utils.current_tai()
+            self.state = SlipRingState.OVER_LOW_LIMIT
+
+        return MAXIMUM_SLIP_RING_POWER_CAPACITY
+
+    async def _handle_over_low_mimit(self, current_power_draw: float) -> float:
```

Comment:
typo in function name? `mimit` -> `limit`?

Reply:
Good catch, thanks!